src/main/java/frc/robot/subsystems/scoring/ShooterIOSim.java

```diff
@@ -0,0 +1,118 @@
+    private final FlywheelSim shooterRightSim =
+            new FlywheelSim(DCMotor.getKrakenX60(1), 0.5, 0.0010639061);
+
+    private final PIDController shooterLeftController =
+            new PIDController(
+                    ScoringConstants.shooterkP,
+                    ScoringConstants.shooterkI,
+                    ScoringConstants.shooterkD);
+    private final PIDController shooterRightController =
+            new PIDController(
+                    ScoringConstants.shooterkP,
+                    ScoringConstants.shooterkI,
+                    ScoringConstants.shooterkD);
+
+    private final SimpleMotorFeedforward shooterFeedforward =
+            new SimpleMotorFeedforward(
+                    ScoringConstants.shooterkS,
+                    ScoringConstants.shooterkV,
+                    ScoringConstants.shooterkA);
+
+    private boolean override = false;
+
+    double shooterLeftGoalVelRPM = 0.0;
+    double shooterLeftAppliedVolts = 0.0;
+
+    double shooterRightGoalVelRPM = 0.0;
+    double shooterRightAppliedVolts = 0.0;
+
+    double kickerVolts = 0.0;
+
+    @Override
+    public void setShooterVelocityRPM(double velocity) {
+        shooterLeftGoalVelRPM = velocity;
+        shooterRightGoalVelRPM = velocity * ScoringConstants.shooterOffsetAdjustment;
+    }
+
+    @Override
+    public void setKickerVolts(double volts) {
+        kickerVolts = volts;
+    }
+
+    @Override
+    public void setOverrideMode(boolean override) {
+        this.override = override;
+    }
+
+    @Override
+    public void setOverrideVolts(double volts) {
+        shooterLeftAppliedVolts = volts;
+        shooterRightAppliedVolts = volts;
+    }
+
+    @Override
+    public void setPID(double p, double i, double d) {
+        shooterLeftController.setPID(p, i, d);
+        shooterRightController.setPID(p, i, d);
+    }
+
+    @Override
+    public void updateInputs(ShooterIOInputs inputs) {
+        shooterLeftSim.update(Constants.loopTime);
+        shooterRightSim.update(Constants.loopTime);
+
+        if (!override) {
+            shooterLeftAppliedVolts =
+                    shooterFeedforward.calculate(shooterLeftSim.getAngularVelocityRadPerSec())
+                            + shooterLeftController.calculate(
+                                    shooterLeftSim.getAngularVelocityRadPerSec(),
+                                    shooterLeftGoalVelRPM
+                                            * ConversionConstants.kRPMToRadiansPerSecond);
+            shooterRightAppliedVolts =
+                    shooterFeedforward.calculate(shooterRightSim.getAngularVelocityRadPerSec())
+                            + shooterRightController.calculate(
+                                    shooterRightSim.getAngularVelocityRadPerSec(),
+                                    shooterRightGoalVelRPM
+                                            * ConversionConstants.kRPMToRadiansPerSecond);
+        }
```

Comment:
Let's move control logic to applyOutputs and migrate any inputs to a new outputs log class